hw2/04-background.js

```diff
@@ -1 +1,59 @@
 // Add your code here
+const START_CLASS = "btn-primary";
+const STOP_CLASS = "btn-danger";
+const BASE_TRANSITION_DURATION = 3000;
+const BASE_BACKGROUND_CLASS = "bg-primary";
+
+const input = document.querySelector("#interval");
+const callToActionButton = document.querySelector("#cta");
+const body = document.querySelector("body");
+body.style.transitionDuration = `${BASE_TRANSITION_DURATION}ms`;
+
+let interval;
+
+const startInterval = () => {
+  body.classList.remove(BASE_BACKGROUND_CLASS);
+  const newIntervalInMs = input.value * 1000;
+
+  interval = setInterval(() => {
+    const newRGBA = `rgba(${getNewRandomColorValue()}, ${getNewRandomColorValue()}, ${getNewRandomColorValue()}, 0.5)`;
+    body.style.backgroundColor = newRGBA;
+
+    body.style.transitionDuration = Math.max(
+      BASE_TRANSITION_DURATION,
+      input.value / 10,
+    );
+  }, Math.max(newIntervalInMs, BASE_TRANSITION_DURATION));
+
+  input.disabled = true;
+  modifyCTA("start");
+};
+
+const stopInterval = () => {
+  clearInterval(interval);
+  input.disabled = false;
+  modifyCTA("stop");
+};
+
+const modifyCTA = (type) => {
+  const isStart = type === "stop";
```

Comment:
Perhaps add parentheses to the condition to make this line a bit more readable, or reformat it to use the ternary  operator like the code below it.

Reply:
sorry I'm not clear on this, can you clarify ❓ 

also, the ternary below is based on `isStart` by the way, so I'm not sure if we can refactor this like you mentioned

Reply:
```suggestion
  const isStart = (type === "stop");
```

Reply:
I don't think prettier would like this unless you have a custom rule. Can you show me what rule you're using ❓

Reply:
If Prettier doesn't accept it that's fine too. It's not a big deal. I just thought it would make it easier to read at a quick glance.

Reply:
gotcha!